pkg/core/native/notary.go

```diff
@@ -162,6 +165,14 @@ func (n *Notary) OnPersist(ic *interop.Context) error {
 func (n *Notary) PostPersist(ic *interop.Context) error {
 	n.lock.Lock()
 	defer n.lock.Unlock()
+	if n.Module != nil {
+		pubs, err := n.GetNotaryNodes(ic.DAO)
+		if err != nil {
+			return err
+		}
+		n.Module.UpdateNotaryNodes(pubs)
```

Comment:
We do because we don't have a cache for notary nodes in native Notary contract, so `n.isValid` is responsible for `n.maxNotValidBeforeDelta` only.

I added a cache for notary nodes to native Designate contract. Also, I moved Notary module to Designate contract, so now we call `UpdateNotaryNodes` only when notary nodes have been changed.